staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go

```diff
@@ -2792,6 +2792,257 @@ func TestToken(t *testing.T) {
+						},
+						{
+							Expression: `user.extra["test.x-kubernetes.io/warnafter"][0] == "1700081020"`,
+						},
+						{
+							Expression: `has(user.extra.authentication__dot__x__dash__kubernetes__dot__io__slash__pod__dash__name)`, // has requires field based access
+						},
+						{
+							Expression: `user.extra[?"test.x-kubernetes.io/missing"].orValue([]) == []`, // optionals can be used even when has() cannot
+						},
+					},
+				},
+				now: func() time.Time { return now },
+			},
+			signingKey: loadRSAPrivKey(t, "testdata/rsa_1.pem", jose.RS256),
+			pubKeys: []*jose.JSONWebKey{
+				loadRSAKey(t, "testdata/rsa_1.pem", jose.RS256),
+			},
+			claims: fmt.Sprintf(`{
+				"aud": [
+					"https://kubernetes.default.svc"
+				],
+				"exp": %d,
+				"iat": 1700077413,
+				"iss": "https://kubernetes.default.svc",
+				"jti": "ea28ed49-2e11-4280-9ec5-bc3d1d84661a",
+				"kubernetes.io": {
+					"namespace": "kube-system",
+					"node": {
+						"name": "127.0.0.1",
+						"uid": "58456cb0-dd00-45ed-b797-5578fdceaced"
+					},
+					"pod": {
+						"name": "coredns-69cbfb9798-jv9gn",
+						"uid": "778a530c-b3f4-47c0-9cd5-ab018fb64f33"
+					},
+					"serviceaccount": {
+						"name": "coredns",
+						"uid": "a087d5a0-e1dd-43ec-93ac-f13d89cd13af"
+					},
+					"warnafter": 1700081020
+				},
+				"nbf": %d,
+				"sub": "system:serviceaccount:kube-system:coredns"
+}`, valid.Unix(), now.Unix()),
+			want: &user.DefaultInfo{
+				Name: "system:serviceaccount:kube-system:coredns",
+				UID:  "a087d5a0-e1dd-43ec-93ac-f13d89cd13af",
+				Groups: []string{
+					"system:serviceaccounts", "system:serviceaccounts:kube-system",
+				},
+				Extra: map[string][]string{
+					"authentication.x-kubernetes.io/pod-name": {
+						"coredns-69cbfb9798-jv9gn",
+					},
+					"authentication.x-kubernetes.io/pod-uid": {
+						"778a530c-b3f4-47c0-9cd5-ab018fb64f33",
+					},
+					"authentication.x-kubernetes.io/node-name": {
+						"127.0.0.1",
+					},
+					"authentication.x-kubernetes.io/node-uid": {
+						"58456cb0-dd00-45ed-b797-5578fdceaced",
+					},
+					"authentication.kubernetes.io/credential-id": {
+						"JTI=ea28ed49-2e11-4280-9ec5-bc3d1d84661a",
+					},
+					"authentication.x-kubernetes.io/credential-id": {
+						"JTI=ea28ed49-2e11-4280-9ec5-bc3d1d84661a",
+					},
+					"test.x-kubernetes.io/warnafter": {
+						"1700081020",
+					},
+					"test.x-kubernetes.io/namespace": {
+						"kube-system",
+					},
+				},
+			},
+		},
+		{
+			name: "claim mappings with expressions and deeply nested claim - success - claims payload with deep use of __dot__",
+			options: Options{
+				JWTAuthenticator: apiserver.JWTAuthenticator{
+					Issuer: apiserver.Issuer{
+						URL:       "https://auth.example.com",
+						Audiences: []string{"my-client"},
+					},
+					ClaimValidationRules: []apiserver.ClaimValidationRule{ // has requires field based access
+						{
+							Expression: `has(claims.turtle.foo.bar.baz.username__dot__io)`,
```

Comment:
Could we structure this to include two nested fields with dots in their names?
Something like: `claims.turtle.foo.bar.baz.username__dot__io.name__dot__io`
That way, we can validate behavior with multi-level nested keys that require unescaping.